### PR TITLE
Various Debian packaging fixes

### PR DESCRIPTION
**Pull request by martinpitt**
Please see individual commits for details.

### CODE REVIEW

**Comment by martinpitt** [comment]
Force-pushed to add "Closes #" to the last commit for push-button merging.

**Comment by dperpeet** [comment]
Commit message requires a fix:
```
tools: Simplify Debian installation

There is no need to override dh_auto_install, the PAM file can be
handled in during dh_install.
```
-->
`handled in dh_install.`

File: tools/debian/rules [hunk 1]
```diff
@@ -43,19 +43,27 @@ override_dh_auto_configure: debian/git-patches-applied
...
 override_dh_systemd_start:
 	dh_systemd_start -p cockpit-ws cockpit.socket
 
 override_dh_install:
-	dh_install --list-missing -Xusr/src/debug
+	# Debian/Ubuntu PAM config
+	mkdir -p debian/tmp/etc/pam.d
+	install -p -m 644 tools/cockpit.debian.pam debian/tmp/etc/pam.d/cockpit
+
+	# don't ship broken branding symlinks
```
Comment (dperpeet) [inline tools/debian/rules]: Should we change the way we do this on the other distros as well? Right now we just accept dangling symlinks...

File: tools/debian/rules [hunk 2]
```diff
@@ -43,19 +43,27 @@ override_dh_auto_configure: debian/git-patches-applied
...
+	for d in rhel fedora centos; do rm -r debian/tmp/usr/share/cockpit/branding/$$d; done
+	dpkg-vendor --derives-from ubuntu || rm -r debian/tmp/usr/share/cockpit/branding/ubuntu
+
+	# unpackaged modules
```
Comment (dperpeet) [inline tools/debian/rules]: Should we document or link this somewhere? Right now we need to update the spec file for rpms and this whenever fiddling with the packages. On the other hand, it's something that we just see fail in tests.

Comment (stefwalter) [inline tools/debian/rules]: The intent is that we have dangling symlinks for branding we cannot ship. The intent is also that we ship all branding in all packaged versions of Cockpit. The same binaries are often used on other distributions. This happens in the cockpit/ws container.

Comment (martinpitt) [inline tools/debian/rules]: If the dangling symlinks are put into a, say, Debian container, their RHEL logo references still won't magically appear. So I don't see much point in shipping those. If you object heavily, I can mark the bug as "wontfix" and ignore the package check.

As for the "unpackaged modules", that indeed needs to be updated if we e. g. add a new cockpit module and *don't* package it for Debian. The purpose of this is exactly this: Add a new module, forget to package it, the test yells at you until you either add the packaging or add it to that explicit blacklist to have a consciuous decision that you don't want it.

File: tools/debian/rules (hunk 1 above)
Comment (martinpitt) [inline tools/debian/rules]: I'm fine with doing the equivalent change (weed out Debian/Ubuntu branding) to the RPMs. (See discussion below)

File: tools/debian/rules (hunk 2 above)
Comment (martinpitt) [inline tools/debian/rules]: Also note that RPMs behave like `--fail-missing` by default. Our spec file already has plenty of rm commands for the exact same purpose.

For symmetry I adjusted the commit to also remove the broken debian/ubuntu branding symlinks from the RPMs. It's fairly safe to say that in the forseeable future Debian won't move to RPM ☺.

Untagging "bot" as this should now run tests on all OSes.

I also fixed the commit message typo.

Comment (stefwalter) [inline tools/debian/rules]: > If the dangling symlinks are put into a, say, Debian container, their RHEL logo references still won't magically appear. So I don't see much point in shipping those. If you object heavily, I can mark the bug as "wontfix" and ignore the package check.

They may not appear for a Debian container ... but for Fedora and RHEL they do. The cockpit/ws container explicitly mounts directories from the host when run on Atomic Host.

So I'm fine with making Debian/Ubuntu an exception to the container insanity here. But there's a reason for shipping the branding/dangling links on Fedora, RHEL, CentOS.

File: tools/cockpit.spec [hunk 3]
```diff
@@ -144,6 +144,10 @@ install -p -m 644 tools/cockpit.pam $RPM_BUILD_ROOT%{_sysconfdir}/pam.d/cockpit
 rm -f %{buildroot}/%{_libdir}/cockpit/*.so
 install -p -m 644 AUTHORS COPYING README.md %{buildroot}%{_docdir}/%{name}/
 
+# Debian/Ubuntu logos are not available on rpm based distros
+rm -rf %{buildroot}/%{_datadir}/%{name}/branding/debian
+rm -rf %{buildroot}/%{_datadir}/%{name}/branding/ubuntu
```
Comment (stefwalter) [inline tools/cockpit.spec]: This is used in ```atomic run``` style containers, or when used with the docker run command that enables. We shouldn't remove the  branding from the RPMs.

Comment (martinpitt) [inline tools/cockpit.spec]: OK, I reverted to the previous version that only removes them from debs.

**Pull request closed without merging**